tests/s3/versioning/test_versioning_multipart.py

```diff
@@ -0,0 +1,213 @@
+        assert_utils.assert_true(resp[0], resp[1])
+        res = self.s3test_obj.create_bucket(self.bucket_name)
+        assert_utils.assert_true(res[0], res[1])
+        assert_utils.assert_equal(res[1], self.bucket_name, res[1])
+        self.log.info("Created a bucket with name : %s", self.bucket_name)
+        self.log.info("ENDED: Setup operations.")
+        yield
+        self.log.info("STARTED: Teardown operations.")
+        s3ver_cmn_lib.empty_versioned_bucket(self.s3ver_test_obj, self.bucket_name)
+        res = self.s3test_obj.delete_bucket(self.bucket_name, force=True)
+        assert_utils.assert_true(res[0], res[1])
+        if system_utils.path_exists(self.test_dir_path):
+            system_utils.remove_dirs(self.test_dir_path)
+        self.log.info("ENDED: Teardown operations.")
+
+    @CTFailOn(error_handler)
+    def test_preexist_mpu_versioning_enabled_bkt_41284(self):
+        """
+        Test pre-existing multipart upload in a versioning enabled bucket
+        """
+        self.log.info("STARTED: Test pre-existing multipart upload in a versioning enabled bucket")
+        self.log.info("Step 2-4: Upload multipart object to a bucket")
+        versions = {}
+        s3ver_cmn_lib.upload_version(self.s3mp_test_obj, self.bucket_name, self.object_name,
+                                     self.test_file_path, versions_dict=versions,
+                                     is_multipart=True, total_parts=2, file_size=10,
+                                     is_unversioned=True)
+        self.log.info("Step 5: Put bucket versioning with status as Enabled")
+        res = self.s3ver_test_obj.put_bucket_versioning(bucket_name=self.bucket_name)
+        assert_utils.assert_true(res[0], res[1])
+        self.log.info("Step 6: List Object Versions")
+        s3ver_cmn_lib.check_list_object_versions(self.s3ver_test_obj, bucket_name=self.bucket_name,
+                                                 expected_versions=versions)
+        self.log.info("Step 7: Check GET/HEAD Object")
+        s3ver_cmn_lib.check_get_head_object_version(self.s3test_obj, self.s3ver_test_obj,
+                                                    bucket_name=self.bucket_name,
+                                                    object_name=self.object_name)
+        self.log.info("Step 8: Check GET/HEAD Object with versionId = null")
+        s3ver_cmn_lib.check_get_head_object_version(self.s3test_obj, self.s3ver_test_obj,
+                                                    bucket_name=self.bucket_name,
+                                                    object_name=self.object_name,
+                                                    version_id="null")
+        self.log.info("ENDED: Test pre-existing multipart upload in a versioning enabled bucket")
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-41285")
+    @CTFailOn(error_handler)
+    def test_mpu_ver_enabled_bkt_41285(self):
+        """
```

Comment:
Make docstring single liner if within 100 chars. It will reduce number of lines from module,